docker-compose.yml

```diff
@@ -1,38 +1,81 @@
 version: "3"
 
 services:
+  redis:
+    image: redis:5
+    restart: unless-stopped
+    ports:
+      - "6379:6379"
+    healthcheck:
+      test: ['CMD', 'redis-cli', 'ping']
+      interval: 1s
+      timeout: 3s
+      retries: 30
+
+  elasticsearch:
+    image: kuzzleio/elasticsearch:7
+    restart: unless-stopped
+    ports:
+      - "9200:9200"
+    environment:
+      - ingest.geoip.downloader.enabled=false
+    healthcheck:
+      test: ['CMD', 'curl', '-f', 'http://localhost:9200']
+      interval: 2s
+      timeout: 2s
+      retries: 10
+    ulimits:
+      nofile: 65536
+    volumes:
+      - esdata01:/usr/share/elasticsearch/data
+
   kuzzle:
-    image: kuzzleio/kuzzle-runner:14
-    command: ${DOCKER_COMMAND:-npm run dev}
+    image: kuzzleio/kuzzle-runner:16.18.0
+    restart: unless-stopped
+    command: "bash /docker/start-backend.sh"
+    profiles: ["backend"]
+    depends_on:
+      redis:
+        condition: service_healthy
+      elasticsearch:
+        condition: service_healthy
     cap_add:
       - SYS_PTRACE
     ulimits:
       nofile: 65536
     sysctls:
       - net.core.somaxconn=8192
-    volumes:
-      - .:/var/app
     ports:
       - "7512:7512"
       - "9229:9229"
       - "1883:1883"
-    depends_on:
-      - redis
+    healthcheck:
+      test: ['CMD', 'curl', '-f', 'http://kuzzle:7512/_healthcheck']
+      timeout: 1s
+      interval: 2s
+      retries: 10
+      start_period: 30s
```

Comment:
I think we should not set a Docker Compose healthcheck on Kuzzle since functional test for example could lead to permissions issues on this route

Reply:
Or at least you should handle 403/401 status code to be allowed

Reply:
The healthcheck route are not forced to be public, or in minimal case, always allow call from localhost ?
Also, I don't see how to handle 403/401 status code here

Reply:
The only way to do that is to create a custom healthcheck (like a Node.js or bash script for example)but for now ignore my comment

Reply:
This health check method is public now 👌🏼 we should not have any issue see https://github.com/kuzzleio/kuzzle/pull/2455